manifest.json

```diff
@@ -1,20 +1,28 @@
 {
-  "description": "A basic template for a Mozilla Rally Study.",
-  "author": "Mozilla",
+  "description": "Local News Demand Study",
+  "author": "Stanford GSB",
   "manifest_version": 2,
-  "name": "Rally Basic Study Template",
-  "version": "0.3.1",
-  "homepage_url": "https://github.com/mozilla-ion/ion-basic-study",
-
-  "applications": {
-    "gecko": {
-      "id": "ion-basic-study@mozilla.org",
-      "strict_min_version": "77.0a1"
-    }
-  },
-
-  "permissions": [],
+  "name": "Local News Demand Study",
+  "version": "0.1",
+  "homepage_url": "https://github.com/bchivers-stanford/local_news_demand",
 
+  "permissions": [
+    "<all_urls>",
+    "idle",
+    "notifications",
+    "storage",
+    "tabs",
+    "unlimitedStorage",
+    "webRequest",
+    "alarms",
+    "telemetry",
+    "history",
+    "webNavigation",
+    "webRequest",
+    "webRequestBlocking"
+  ],
+  "browser_action": {},
+  "content_security_policy": "script-src 'self' 'sha256-csyiOLMfXk2f5pU99mqYFyshgnVYbdp6o9bnQ9hntPA=' 'sha256-nYNRfLKTaKqgi4+CK/mcp9hdSsmD8F17GWuo+vQGfqU='; object-src 'self'",
```

Comment:
This CSP allows execution of remote code, which is forbidden. What's the purpose of adding these hashes? Can we remove them?

Reply:
@jonathanmayer, would you mind commenting on this ^?

Reply:
Context that might be helpful: This CSP hash is necessary because of WebScience, the library for building Rally studies. WebScience provides basic survey functionality in the `userSurvey` module.

https://github.com/mozilla-rally/web-science/blob/main/src/userSurvey.js
https://github.com/mozilla-rally/web-science/blob/main/src/html/userSurvey.popupPrompt.html
https://github.com/mozilla-rally/web-science/blob/main/src/html/userSurvey.popupNoPrompt.html

The survey UX involves a WebExtensions browser action, and the content of that browser action is HTML and JS. We considered a few options for how to load that HTML and JS into Rally studies.

* Require study developers to copy the relevant WebScience file(s) into the study extension. We decided against this approach, because study developers could easily mess up file paths or versioning. A special Rollup plugin for WebScience could handle file management, but then we'd be maintaining and requiring a Rollup plugin.

* Inline the survey HTML and JS in WebScience, so that Rally study developers don't have to manage files. Study developers would then provide CSP authorization for `blob:` to load the inlined content. We decided against this approach, too, because of the significant security implications—any blob URL, even from outside the study add-on, could pass the CSP.

* Inline the survey HTML and JS in WebScience, then use script hashes in the CSP so that only our survey content is permitted. That's where we landed.

The fundamental issue is that, if a WebExtensions browser action loads content from the add-on rather than a remote page, the add-on has no way of specifying a CSP for just the browser action.

Reply:
Can we please use an approach that doesn't require to relax the CSP, especially for scripts? Thanks!

Reply:
@jonathanmayer changed the way this works in WebScience https://github.com/mozilla-rally/web-science/pull/102 - once we merge that, these can be loaded as standalone `file:`s so we don't need to fiddle with CSP anymore.

@bchivers-stanford, re: https://github.com/rhelmer/local_news_demand/pull/1#discussion_r645099553 let's figure out how to help you align with this approach, so we can do it in a way add-ons reviewers are happy with.